l2lsp/identify_method.go

```diff
@@ -0,0 +1,32 @@
+package l2lsp
+
+import (
+	"github.com/HexmosTech/lama2/l2lsp/lsp_req"
+	"github.com/HexmosTech/lama2/l2lsp/lsp_res"
+	"github.com/HexmosTech/lama2/l2lsp/methods"
+)
+
+var isShutdownRequested bool
+
+func HandleMethod(request lsp_req.JSONRPCRequest) lsp_res.JSONRPCResponse {
+	if isShutdownRequested && request.Method != "exit" {
+		return lsp_res.InvalidReqAfterShutdown(request)
+	}
+
+	switch request.Method {
+	case "initialize":
+		return methods.Initialize(request)
+
+	case "shutdown":
+		return methods.Shutdown(request, isShutdownRequested)
+
+	case "exit":
```

Comment:
what is the difference between exit and shut down?

Reply:
* shutdown: Graceful termination, executes all leftover requests, doesn't take any more incoming reqs.
* exit: Kill the process with os.exit(0)
* although these are not implemented as of now, but extensible and once we have state management we will need this.

Reply:
Worth adding a comment in code.

Reply:
right